cargo/cargo_build_script.bzl

```diff
@@ -5,16 +5,33 @@ load("@io_bazel_rules_rust//rust:private/utils.bzl", "find_toolchain")
 load("@io_bazel_rules_rust//rust:rust.bzl", "rust_binary")
 
 def _expand_location(ctx, env, data):
-    for directive in "$(execpath ", "$(location ":
+    """[summary]
```

Comment:
```suggestion
    """Performs location-macro expansion on string values.
    
    Note that exec-root relative locations will be exposed to the build script as absolute paths, rather than the ordinary exec-root relative paths, because cargo build scripts do not run in the exec root.
```

Reply:
Thanks for filling out those docs. Just a minor clarification on the first rustc comment  below - the docs should recommend only using rootpath in that case, as execpath will fail in the case of generated sources (can be reproduced by changing the second reference to GENERATED_DATA or SOME_TOOL in examples/env_locations/BUILD)